### PR TITLE
Initial Audit Log DB Layer + Tests

### PR DESCRIPTION
**Pull request by gbdubs**

### CODE REVIEW

File: db/sqldb/sqldb_test.go [hunk 1]
```diff
@@ -83,6 +84,7 @@ func TestSchemaHistory(t *testing.T) {
 		{ID: 1, Version: 1}, // 0001_create_schema_migrations_history
 		{ID: 2, Version: 2}, // 0002_create_user_table
 		{ID: 3, Version: 3}, // 0003_domain_types
+		{ID: 4, Version: 4}, // 0003_audit_log_tweaks
```
Comment (bcspragu) [inline db/sqldb/sqldb_test.go:87]:
```suggestion
		{ID: 4, Version: 4}, // 0004_audit_log_tweaks
```

File: db/sqldb/user.go [hunk 2]
```diff
@@ -75,13 +75,18 @@ func (d *DB) CreateUser(tx db.Tx, u *pacta.User) (pacta.UserID, error) {
 	if err := validateUserForCreation(u); err != nil {
 		return "", fmt.Errorf("validating user for creation: %w", err)
 	}
+	var pl *string
```
Comment (bcspragu) [inline db/sqldb/user.go:78]: Same nit, I feel icky about pointer primitives when our SQL driver has dedicated types for this

**Review by bcspragu — APPROVED** [review]

File: db/sqldb/audit_log.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+	audit_log.primary_target_owner_id,
+	audit_log.secondary_target_type,
+	audit_log.secondary_target_id,
+	audit_log.secondary_target_owner_id,
+	audit_log.created_at
+`
+
+func (d *DB) AuditLogs(tx db.Tx, q *db.AuditLogQuery) ([]*pacta.AuditLog, *db.PageInfo, error) {
+	if q.Limit <= 0 {
+		return nil, nil, fmt.Errorf("limit must be greater than 0, was %d", q.Limit)
```
[the remark this answers is not in the document]
Comment (gbdubs) [reply db/sqldb/audit_log.go:32]: Limit must be greater than zero.

File: db/sqldb/audit_log.go [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+		&a.CreatedAt,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("scanning into audit_log: %w", err)
+	}
+	if t, err := pacta.ParseAuditLogActorType(actorType); err != nil {
+		return nil, fmt.Errorf("parsing audit_log actor_type: %w", err)
+	} else {
+		a.ActorType = t
+	}
```
[the remark this answers is not in the document]
Comment (gbdubs) [reply db/sqldb/audit_log.go:113]: GC

File: db/sqldb/user.go (hunk 2 above)
Comment (gbdubs) [reply db/sqldb/user.go:78]: SG

**Pull request merged**